# ROS topic and bagfile names
bagfile_name: $(arg bagfile_name)
odom_topic: /husky_velocity_controller/odom
rs_odom_topic: /camerat265/odom/sample
fix_topic: /fix
tf_topic: /tf
depth_img_topic: /zed/zed_node/depth/depth_registered
left_img_topic: /zed/zed_node/left_raw/image_raw_color/compressed
#left_img_topic: /zed/zed_node/left/image
detections_topic: /tpu/detections
pcl_topic: /velodyne_points


# General system settings
system:
  # - System flags
  # ---------------------
  debug: True
  # ---- Define system inputs
  use_planes: True
  use_ground_plane: True
  use_landmarks: False
  use_corners: True
  use_icp: False
  use_gps: False
  # ---------------------
  # GPS datum
  gps_datum:
    lat: 41.6520631667
    long: -8.7052965
  # ---------------------
  # Sensors to base static tfs (x, y, z, qx, qy, qz, qw)
  cam2base: [0.343, 0.079, 0.820, -0.002, 0.100, -0.004, 0.995]
  vel2base: [0.000, 0.000, 0.942, -0.000, 0.000, -0.000, 1.000]

# Camera calibration info
camera_info:
  baseline: 0.12       # meters
  depth_hfov: 90.0     # degrees - depth image horizontal field of view
  depth_vfov: 60.0     # degrees - depth image vertical field of view
  img_width: 1280      # pixels
  img_height: 720      # pixels
  fx: 692.95849609375  # pixels
  fy: 636.41424560546  # pixels
  cx: 692.95849609375  # pixels
  cy: 382.80633544921  # pixels


multilayer_mapping:
  # Grid map definitions
  grid_map:
    origin:
      x: -75
      y: -75
    width: 150
    height: 150
    resolution: 0.25 # meters
    metric: "euclidean"
    save_map: True
    output_folder: "/home/andre-criis/Source/vineslam_data/raw_data/maps/"
    input_file: "/home/andre-criis/map.xml"
  # Feature extractor and descriptor settings
  image_feature:
    hessian_threshold: 3500 # SURF threshold: the higher the value, the lesser the number of features computed
  # 3D cloud feature extraction settings
  cloud_feature:
    downsample_factor: 1  # constant to downscale dense disparity image
  # 3D map definitions
  map_3D:
    max_range: 5.0 # meters
    max_height: 4.5 # meters
  # ICP definitions
  ICP:
    max_iters: 20
    distance_threshold: 0.02
    reject_outliers: True


pf:
  n_particles: 500

  srr: 0.005 # rotation - rotation (rad / rad)
  srt: 0.100 # rotation - translation (rad / meter)
  str: 0.008 # translation - rotation (meter / rad)
  stt: 0.100 # translation - translation (meter / meter)

  ## --- PARTICLES INNOVATION -- #
  sigma_xy: 0.15 # meters (just for initialization)
  sigma_z: 0.01 # meters
  sigma_roll: 0.01 # degrees
  sigma_pitch: 0.01 # degrees
  sigma_yaw: 1.0 # degrees (just for initialization)

  ## --- PARTICLES UPDATE -- #
  sigma_landmark_matching: 0.01 # meters
  sigma_feature_matching: 0.05 # meters
  sigma_corner_matching: 0.05 # meters
  sigma_planes_yaw: 1.0 # degrees
  sigma_ground_rp: 1.0 # degrees
  sigma_gps: 0.08 # meters

  k_clusters: 3